File: Cootie/cootie.py
"""
Coupon collector example modified to simulate the dice
game "Cootie."
https://en.wikipedia.org/wiki/The_Game_of_Cootie
"""
__author__ = "David Owen and Noah Barrall"

from sys import argv
from random import randrange

if __name__ == "__main__":
    trials = 1000
    totalTries = 0

    if len(argv) >= 2: 
        trials = int(argv[1])

    for i in range(trials):
        tries = 0
    
    # body
        while True:
            tries += 1

            if randrange(1, 7) == 1:
                break

                # head
        while True:
            tries += 1

            if randrange(1, 7) == 2:
                break

        # 1 headgear, 1 pair of eyes, 1 tongue, 6 feet
        partsNeeded = [1, 1, 1, 6]
        numNeeded = sum(partsNeeded)
        partsCollected = [0] * 4
        numCollected = 0

        while numCollected < numNeeded:
            tries += 1
            r = randrange(1, 7)

            if r >= 3:
                i = r - 3

                if partsCollected[i] < partsNeeded[i]:
                    numCollected += 1
                    partsCollected[i] += 1
            
        totalTries += tries

    average_tries = totalTries/trials
    print("The average number of tries was: {:.2f}".format(average_tries))

File: Cootie/Cootie.java
/**
*
* Cootie Program written in Java
*
* @author Noah Barrall
*/

public class Cootie {
    public static void main(String[] args) {
        int trials = 1000;
        int totalTries = 0;

        if (args.length >= 1) {
            
            trials = Integer.parseInt(args[0]);
        }

        for (int i = 0; i < trials; i++) {
            int tries = 0;
        
            while (true) {
                tries++;
             
               if ((int) (Math.random() * 6) + 1 == 1) {
                    break;
               }
            }

            while (true) {
                tries++;
            
                if ((int)(Math.random() * 6) + 1 == 2) {
                    break;
                }
            }

            int [] partsNeeded = {1, 1, 1, 6};
            int numNeeded = 9;
            int [] partsCollected = new int [4]; 
            int numCollected = 0;

            while (numCollected < numNeeded) {
                tries++;
                int r = (int) (Math.random() * 6 + 1);

                    if (r >= 3) {
                        int j = (r - 3);

                            if (partsCollected[j] < partsNeeded[j]) {
                                numCollected++;
                                partsCollected[j]++;
                            }
                    }
            }
            totalTries += tries;
        }
        double averageTries = (double) totalTries / trials;
        System.out.printf("The average number of tries was %.2f", averageTries);
    }    
}

